src/statistics.jl

```diff
@@ -0,0 +1,88 @@
+abstract type AbstractStatistic end
+
+struct Mean{D<:AbstractDomain} <: AbstractStatistic
+  domain::D
+  values::Dict{Symbol, Vector} 
+end
+
+function Mean(solution::SimulationSolution)
+  # output dictionary
+  values = Dict(var => mean(real) for (var, real) in solution.realizations)
+    
+  Mean{typeof(solution.domain)}(solution.domain, values)
+end
+
+mean(solution::SimulationSolution) = Mean(solution)
+
+struct Variance{D<:AbstractDomain} <: AbstractStatistic
+  domain::D
+  values::Dict{Symbol, Vector} 
+end
+
+function Variance(solution::SimulationSolution)
+  # output dictionary
+  values = Dict(var => var(real) for (var, real) in solution.realizations)
+    
+  Variance{typeof(solution.domain)}(solution.domain, values)
+end
+
+var(solution::SimulationSolution) = Variance(solution)
+
+struct Quantile{D<:AbstractDomain, P<:Real}
```

Comment:
I don't think we need  to generalize on `P`, we can always use `Float64` so this type parameter is not necessary. It is making the quantile type inconsistent with the other types that are only parameterized by `D`.